include/uxr/agent/utils/SeqNum.hpp

```diff
@@ -30,18 +31,40 @@ class SeqNum
     /*
      * Operators.
      */
-    operator uint16_t() const { return seq_num_; }
-    operator int() const { return seq_num_; }
+    inline operator uint16_t() const { return seq_num_; }
+
+    friend bool operator<(const SeqNum& lhs, const SeqNum& rhs)
+    {
+        return (lhs.seq_num_ != rhs.seq_num_) &&
+               (((lhs.seq_num_ < rhs.seq_num_) && ((rhs.seq_num_ - lhs.seq_num_) < (ADD_RANGE[1] + 1))) ||
+                ((lhs.seq_num_ > rhs.seq_num_) && ((lhs.seq_num_ - rhs.seq_num_) > (ADD_RANGE[1] + 1))));
+    }
+
+    friend bool operator>(const SeqNum& lhs, const SeqNum& rhs) { return rhs < lhs; }
+    friend bool operator<=(const SeqNum& lhs, const SeqNum& rhs) { return !(lhs > rhs); }
+    friend bool operator>=(const SeqNum& lhs, const SeqNum& rhs) { return !(lhs < rhs); }
+    friend bool operator==(const SeqNum& lhs, const SeqNum& rhs) { return lhs.seq_num_ == rhs.seq_num_; }
+    friend bool operator==(const SeqNum& lhs, const int& rhs) { return lhs.seq_num_ == uint16_t(rhs); }
+    friend bool operator==(const int& lhs, const SeqNum& rhs) { return uint16_t(lhs) == rhs.seq_num_; }
+    friend bool operator==(const SeqNum& lhs, const uint16_t& rhs) { return lhs.seq_num_ == rhs; }
+    friend bool operator==(const uint16_t& lhs, const SeqNum& rhs) { return lhs == rhs.seq_num_; }
```

Comment:
Sometimes is better to copy integer values instead of const& them.

Reply:
Yes, you are right!